include/nbl/builtin/hlsl/glsl_compat/core.hlsl

```diff
@@ -18,42 +18,42 @@ namespace glsl
 template<typename T>
 T atomicAdd(NBL_REF_ARG(T) ptr, T value)
 {
-    return spirv::atomicAdd<T>(ptr, 1, 0, value);
+    return spirv::atomicAdd<T>(ptr, spv::ScopeDevice, spv::GroupOperationReduce, value);
 }
 template<typename T>
 T atomicAnd(NBL_REF_ARG(T) ptr, T value)
 {
-    return spirv::atomicAnd<T>(ptr, 1, 0, value);
+    return spirv::atomicAnd<T>(ptr, spv::ScopeDevice, spv::GroupOperationReduce, value);
```

Comment:
Read The Spec: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpAtomicAnd

`spirv::atomicAdd` is obviously `OpAtomicAdd`, it takes a Memory Scope (in that enum 1 is ScopeDevice so all good), and Memory Semantics, 0 is None a.k.a. Relaxed

`GroupOperationReduce` is for subgroup intrinsics (also known as Wave Ops